backend/src/main/java/com/bakdata/conquery/util/ConqueryEscape.java

```diff
@@ -2,65 +2,85 @@
 
 import java.io.ByteArrayOutputStream;
 import java.nio.charset.StandardCharsets;
-
-import org.apache.commons.lang3.ArrayUtils;
+import java.util.BitSet;
+import java.util.stream.IntStream;
 
 import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.ArrayUtils;
 
-
+/**
+ * We use a custom escaping for the parts an {@link com.bakdata.conquery.models.identifiable.ids.IId} consists of.
+ * This way, we ensure, that the {@link com.bakdata.conquery.models.identifiable.ids.IId#JOIN_CHAR} is not confused and that the resulting id
+ * can be used safely in a URL path or query without further encoding.
+ */
+@UtilityClass
 public class ConqueryEscape {
-	
+
 	private static final byte ESCAPER = '$';
-	private static final ConqueryEscape INSTANCE = new ConqueryEscape();
 
-	public static String escape(@NonNull String word) {
-		return INSTANCE.escapeString(word);
+	private static final BitSet dontNeedEncoding;
+
+
+	static {
+		// adapted from java.net.UrlEncoder
+		dontNeedEncoding = new BitSet(256);
+
+		IntStream.range('a', 'z').forEach(dontNeedEncoding::set);
+		IntStream.range('A', 'Z').forEach(dontNeedEncoding::set);
+		IntStream.range('0', '9').forEach(dontNeedEncoding::set);
+
+		// We have only a few characters that won't encode unlike UrlEncode
+		dontNeedEncoding.set('-');
+		dontNeedEncoding.set('_');
+
+		// Ensure that occurrences of the escaper are escaped
+		assert !dontNeedEncoding.get(ESCAPER);
+
 	}
-		
-	protected String escapeString(String word) {
-		if(word.isEmpty()) {
+
+	public static String escape(@NonNull String word) {
+		if (word.isEmpty()) {
 			return word;
 		}
-		
-		byte[] bytes = word.getBytes(StandardCharsets.UTF_8);
-		
-		//if the first does not match we escape everything
-		if(!matchesFirst(bytes[0])) {
-			ByteArrayOutputStream baos = new ByteArrayOutputStream(bytes.length + 5);
-			encode(bytes[0], baos);
-			return escapeRequired(bytes, 1, baos);
-		}
-		
-		//if the first matched we walk through the rest and check if any don't match the allowed characters
-		for(int i=1;i<bytes.length;i++) {
-			if(!matchesOther(bytes[i])) {
+
+
+		// start escaping if a caracter is found that needs escaping
+		for (int i = 0; i < word.length(); i++) {
+			final char c = word.charAt(i);
+
+			// Check if the character is larger than a byte or if that byte needs encoding
+			if (((c >> 8) & 0xFF) != 0 || !dontNeedEncoding((byte) (c & 0x00FF))) {
```

Comment:
Bin gerade zu doof das zu übersetzen, aber kannst du das nicht in ein `c >= X` umschreiben? Ich finde die byte schreibweise relativ schwer zu lesen ehrlich gesagt